frontend-wasm/src/code_translator/mod.rs

```diff
@@ -0,0 +1,1035 @@
+        Operator::Br { relative_depth } => translate_br(state, relative_depth, builder, span),
+        Operator::BrIf { relative_depth } => translate_br_if(*relative_depth, builder, state, span),
+        Operator::BrTable { targets } => translate_br_table(targets, state, builder, span)?,
+        Operator::Return => translate_return(state, builder, diagnostics, span)?,
+        /************************************ Calls ****************************************/
+        Operator::Call { function_index } => {
+            translate_call(state, builder, function_index, mod_info, span, diagnostics)?;
+        }
+        /******************************* Memory management *********************************/
+        Operator::MemoryGrow { .. } => {
+            let arg = state.pop1();
+            state.push1(builder.ins().mem_grow(arg, span));
+        }
+        Operator::MemorySize { .. } => {
+            // Return total Miden memory size
+            state.push1(builder.ins().i32(mem_total_pages(), span));
+        }
+        /******************************* Load instructions ***********************************/
+        Operator::I32Load8U { memarg } => {
+            translate_load_zext(I8, I32, memarg, state, builder, span)
+        }
+        Operator::I32Load16U { memarg } => {
+            translate_load_zext(I16, I32, memarg, state, builder, span)
+        }
+        Operator::I32Load8S { memarg } => {
+            translate_load_sext(I8, I32, memarg, state, builder, span);
+        }
+        Operator::I32Load16S { memarg } => {
+            translate_load_sext(I16, I32, memarg, state, builder, span);
+        }
+        Operator::I64Load8U { memarg } => {
+            translate_load_zext(I8, I64, memarg, state, builder, span)
+        }
+        Operator::I64Load16U { memarg } => {
+            translate_load_zext(I16, I64, memarg, state, builder, span)
+        }
+        Operator::I64Load8S { memarg } => {
+            translate_load_sext(I8, I64, memarg, state, builder, span);
+        }
+        Operator::I64Load16S { memarg } => {
+            translate_load_sext(I16, I64, memarg, state, builder, span);
+        }
+        Operator::I64Load32S { memarg } => {
+            translate_load_sext(I32, I64, memarg, state, builder, span)
+        }
+        Operator::I64Load32U { memarg } => {
+            translate_load_zext(I32, I64, memarg, state, builder, span)
+        }
+        Operator::I32Load { memarg } => translate_load(I32, memarg, state, builder, span),
+        Operator::I64Load { memarg } => translate_load(I64, memarg, state, builder, span),
+        /****************************** Store instructions ***********************************/
+        Operator::I32Store { memarg } => translate_store(I32, memarg, state, builder, span),
+        Operator::I64Store { memarg } => translate_store(I64, memarg, state, builder, span),
+        Operator::I32Store8 { memarg } | Operator::I64Store8 { memarg } => {
```

Comment:
This should use `U8` rather than `I8`, since presumably it's intended for storing bytes, not signed integers. Same for the 16 and 32-bit variants below.

Reply:
Done.